### PR TITLE
test: add separate test files for transform mode and feature testing

### PR DESCRIPTION
**Pull request by mertcanaltin**
Add separate test files for transform mode and feature testing [46](https://github.com/nodejs/amaro/issues/46)

- Created `transform-mode.test.js` to test general transformation mode functionalities, including async/await and ES6 module transformations.
- Added `transform-feature.test.js` to verify handling of specific ECMAScript features such as class fields and private class fields.
- Introduced `transform-decorator-namespace.test.js` to test TypeScript-specific features like decorators and namespaces.

### CODE REVIEW

File: test/transform-decorator-namespace.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          return lettersRegexp.test(s);
+        }
+      }
+    }
+  `;
+	const { code } = transformSync(inputCode, {
+		mode: "transform",
+		sourceMaps: true,
+	});
+	t.assert.snapshot(code);
```
Comment (marco-ippolito) [inline test/transform-decorator-namespace.test.js:54]: we should also check the code is runnable

File: test/transform-feature.test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		__dirname,
+		"snapshots",
+		`${path.basename(testPath)}.snapshot`,
+	);
+});
+
+test("should handle class fields", (t) => {
+	const inputCode = `
+    class Counter {
+      count = 0;
```
Comment (marco-ippolito) [inline test/transform-feature.test.js:16]: isnt this valid js code?

File: test/transform-feature.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	const { code } = transformSync(inputCode, {
+		mode: "transform",
+		sourceMaps: true,
+	});
+	t.assert.snapshot(code);
+});
+
+test("should handle nullish coalescing", (t) => {
+	const inputCode = `
+    const value = null ?? 'default';
```
Comment (marco-ippolito) [inline test/transform-feature.test.js:50]: this is alread valid in javascript

**Review by marco-ippolito — CHANGES_REQUESTED** [review]
Please remove tests that are not meaningful

File: test/transform-decorator-namespace.test.js (hunk 1 above)
Comment (mertcanaltin) [reply test/transform-decorator-namespace.test.js]: I wonder if you have a suggestion for this place

File: test/transform-decorator-namespace.test.js [hunk 4]
```diff
@@ -25,15 +26,27 @@ test("should handle class decorators", (t) => {
...
-	t.assert.snapshot(code);
+
+	const snapshotPath = resolveSnapshotPath(
+		"should transform TypeScript class decorators",
+	);
+	if (!fs.existsSync(snapshotPath)) {
+		fs.writeFileSync(snapshotPath, code, "utf-8");
+	}
+
+	const expectedCode = fs.readFileSync(snapshotPath, "utf-8");
```
Comment (marco-ippolito) [inline test/transform-decorator-namespace.test.js:45]: why `fs.writeFile` ? We can use snapshots

File: test/transform-decorator-namespace.test.js (hunk 1 above)
Comment (marco-ippolito) [reply test/transform-decorator-namespace.test.js:63]: You can rebase and check other tests that use the `vm` module

Comment (mertcanaltin) [reply test/transform-decorator-namespace.test.js]: thank you for your suggestion, I have posted a commit I am trying to implement this

**Comment by mertcanaltin** [comment]
Could it be that we are getting this error because we do not support it?

```
test at test/transform-decorator-namespace.test.js:65:1
✖ should transform TypeScript namespaces with additional functionality (4.865125ms)
  '  x TypeScript namespace declaration is not supported in strip-only mode\n    ,-[2:1]\n  1 |     \n  2 | ,->   namespace Validation {\n  3 | |       export interface StringValidator {\n  4 | |         isAcceptable(s: string): boolean;\n  5 | |       }\n  6 | |       const lettersRegexp = /^[A-Za-z]+$/;\n  7 | |       export class LettersOnlyValidator implements StringValidator {\n  8 | |         isAcceptable(s: string) {\n  9 | |           return lettersRegexp.test(s);\n 10 | |         }\n 11 | |         static createValidator(): LettersOnlyValidator {\n 12 | |           return new LettersOnlyValidator();\n 13 | |         }\n 14 | |       }\n 15 | `->   }\n 16 |     \n 17 |       const validator = Validation.LettersOnlyValidator.createValidator();\n 18 |       const isValid = validator.isAcceptable("test");\n    `----\n'`
```

**Comment by marco-ippolito** [comment]
> Could it be that we are getting this error because we do not support it?
> 
> ```
> test at test/transform-decorator-namespace.test.js:65:1
> ✖ should transform TypeScript namespaces with additional functionality (4.865125ms)
>   '  x TypeScript namespace declaration is not supported in strip-only mode\n    ,-[2:1]\n  1 |     \n  2 | ,->   namespace Validation {\n  3 | |       export interface StringValidator {\n  4 | |         isAcceptable(s: string): boolean;\n  5 | |       }\n  6 | |       const lettersRegexp = /^[A-Za-z]+$/;\n  7 | |       export class LettersOnlyValidator implements StringValidator {\n  8 | |         isAcceptable(s: string) {\n  9 | |           return lettersRegexp.test(s);\n 10 | |         }\n 11 | |         static createValidator(): LettersOnlyValidator {\n 12 | |           return new LettersOnlyValidator();\n 13 | |         }\n 14 | |       }\n 15 | `->   }\n 16 |     \n 17 |       const validator = Validation.LettersOnlyValidator.createValidator();\n 18 |       const isValid = validator.isAcceptable("test");\n    `----\n'`
> ```

It says that namespaces are not supported in strip mode, you need to emable transform mode

**Comment by mertcanaltin** [comment]
updated, thanks

**Comment by marco-ippolito** [comment]
what is the difference between `transform-mode.test.js` and `transform-feature.test.js` ? why two separate files for the same thing

File: test/transform-mode.test.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+// Set the path for the snapshots directory
+snapshot.setResolveSnapshotPath((testPath) => {
+	return path.join(
+		__dirname,
+		"snapshots",
+		`${path.basename(testPath)}.snapshot`,
+	);
+});
+
+test("should handle TypeScript decorators", (t) => {
```
Comment (marco-ippolito) [inline test/transform-mode.test.js:14]: isnt this already handled in the `transform-decoratorna-names` test?

Comment (mertcanaltin) [reply test/transform-mode.test.js:14]: Yeah, I think I copied this

**Comment by mertcanaltin** [comment]
> what is the difference between `transform-mode.test.js` and `transform-feature.test.js` ? why two separate files for the same thing

test/transform-mode.test.js I removed the file because it is similar as you said

File: test/transform-decorator-namespace.test.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+			"report",
+			"Report type should be 'report'",
+		);
+		assert.strictEqual(
+			context.report.title,
+			"Test",
+			"Report title should be 'Test'",
+		);
+	} catch (err) {
+		console.error("Error executing script:", err);
```
Comment (marco-ippolito) [inline test/transform-decorator-namespace.test.js:64]: why catch error during test?

File: test/transform-feature.test.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      getCount(): number {
+        return this.#count;
+      }
+    }
+  `;
+	const { code } = transformSync(inputCode, {
+		mode: "transform",
+		sourceMaps: true,
+	});
+	t.assert.snapshot(code);
```
Comment (marco-ippolito) [inline test/transform-feature.test.js:45]: why no execution test here with vm?

File: test/transform-decorator-namespace.test.js (hunk 6 above)
Comment (mertcanaltin) [reply test/transform-decorator-namespace.test.js:64]: Sorry, I was logging the test and forgot to remove it

File: test/transform-feature.test.js (hunk 7 above)
Comment (mertcanaltin) [reply test/transform-feature.test.js:45]: I wonder if there is a problem if I add all the tests on this page

Comment (marco-ippolito) [reply test/transform-feature.test.js:45]: its ok

Comment (mertcanaltin) [reply test/transform-feature.test.js:60]: ı added, thanks

**Review by marco-ippolito — APPROVED** [review]

**Comment by marco-ippolito** [comment]
can you squash please

**Comment by marco-ippolito** [comment]
actually can you move all your new tests into `transform.test.js`. It already exists

**Comment by mertcanaltin** [comment]
Moved here
https://github.com/nodejs/amaro/pull/61

**Pull request closed without merging**